packages/core/src/embeddings/types.ts

```diff
@@ -21,4 +28,30 @@ export abstract class BaseEmbedding {
 
   abstract getTextEmbedding(text: string): Promise<number[]>;
   abstract getQueryEmbedding(query: string): Promise<number[]>;
+
+  async getTextEmbeddingBatch(texts: string[]): Promise<Array<Embedding>> {
+    const resultEmbeddings: Array<Embedding> = [];
+
+    const queue = texts;
+    let batchIndex = 0;
+
+    await BlueBird.map(
```

Comment:
I think BlueBird here is not necessary?

You can do the same thing using Promise.all with array function

```ts
let queue = [];
let idx = 0;
for (const text of texts) {
  queue.push(this.getTextEmbedding(text).then(embedding => {
    console.log('xxx');
    idx++;
    return embedding;
  }));
}
return Promise.all(queue)
```

Reply:
@himself65 Promise.all would try to embed all the texts at once and then get rate limited, so that's why I like to use the Bluebird concurrency

Reply:
@himself65 applied some changes, did a bit different but you're right wouldn't be benefitial for now